src/op_analytics/cli/subcommands/pulls/agora/data_access.py

```diff
@@ -0,0 +1,237 @@
+import glob
+import os
+from enum import Enum
+from functools import cache
+import re
+import polars as pl
+import pyarrow as pa
+
+from op_analytics.coreutils.duckdb_inmem import init_client
+from op_analytics.coreutils.env.aware import OPLabsEnvironment, current_environment
+from op_analytics.coreutils.logger import structlog
+from op_analytics.coreutils.partitioned import (
+    DataLocation,
+    DataWriter,
+    ExpectedOutput,
+    OutputData,
+    SinkMarkerPath,
+    SinkOutputRootPath,
+)
+from op_analytics.coreutils.partitioned.breakout import breakout_partitions
+from op_analytics.coreutils.partitioned.dataaccess import DateFilter, MarkerFilter, init_data_access
+from op_analytics.coreutils.path import repo_path
+from op_analytics.coreutils.time import date_fromstr, now_friendly_timestamp
+from op_analytics.datapipeline.utils.daterange import DateRange
+
+log = structlog.get_logger()
+
+MARKERS_TABLE = "daily_data_markers"
+
+
+class Agora(str, Enum):
+    """Supported agora datasets."""
+
+    DELEGATES = "delegates_v1"
```

Comment:
wanted to better understand the reason to split the data into with/without voting power. Would it be possible to keep voting power as a column and have a single dataset?

Reply:
I had this thought initially because of the assymetric nature of the data (very few with voting power vs the rest), and how we would need to pull the entire table and filter for voting power each time we want to do an analysis.

On a second thought we can leave this out, since it now adds complexity that I'm not 100% is needed.